### PR TITLE
Fix ptrace io plugin open

### LINKED ISSUE #15296: BSD builds are broken
```
p/io_ptrace.c:193:25: error: use of undeclared identifier 'PTRACE_GETSIGINFO'
        ret = r_io_ptrace (io, PTRACE_GETSIGINFO, pid, NULL, &sig);
                               ^
1 error generated.
```

### PR DESCRIPTION
**Pull request by thestr4ng3r**
This reverts commit 0f430b587163d2ecb5fe783108261a63cd45f92e.
@XVilka

### CODE REVIEW

**Comment by ret2libc** [comment]
I don't really understand why. Debugger may be broken, but xvilka's PR doesn't seem to be the reason to me.

**Comment by thestr4ng3r** [comment]
XVilka asked me to open this since he had no git access. Do whatever you want with it.

File: libr/io/p/io_ptrace.c [hunk 1]
```diff
@@ -197,9 +189,10 @@ static RIODesc *__open(RIO *io, const char *file, int rw, int mode) {
 
 	int pid = atoi (file + 9);
 
-	// Safely check if the PID has already been attached to avoid printing errors
-	// and attempt attaching on failure
-	if (!is_pid_already_attached(io, pid, &sig)) {
+	// Safely check if the PID has already been attached to avoid printing errors.
+	ret = r_io_ptrace (io, PTRACE_GETSIGINFO, pid, NULL, &sig);
+	// Attempt attaching on failure
+	if (-1 == ret) {
```
Comment (radare) [inline libr/io/p/io_ptrace.c]: Ret is bool should never be -1

Comment (thestr4ng3r) [reply libr/io/p/io_ptrace.c]: No, ret is int and ptrace returns long.

**Comment by thestr4ng3r** [comment]
#15311 is probably better but ¯\_(ツ)_/¯

**Comment by thestr4ng3r** [comment]
This is how it should have been fixed

**Pull request merged**